storage/remote/otlptranslator/prometheus/metric_name_builder.go

```diff
@@ -322,3 +260,72 @@ func removeItem(slice []string, value string) []string {
 	}
 	return newSlice
 }
+
+// BuildMetricName builds a valid metric name but without following Prometheus naming conventions.
+// It doesn't do any character transformation, it only prefixes the metric name with the namespace, if any,
+// and adds metric type suffixes, e.g. "_total" for counters and unit suffixes.
+//
+// Differently from BuildCompliantMetricName, it doesn't pre-check the presence of unit and type suffixes.
+// If "addMetricSuffixes" is true, it will add them anyway.
+//
+// Please use BuildCompliantMetricName for a metric name that follows Prometheus naming conventions.
+func BuildMetricName(metric pmetric.Metric, namespace string, addMetricSuffixes bool) string {
+	metricName := metric.Name()
+
+	if namespace != "" {
+		metricName = namespace + "_" + metricName
+	}
+
+	if addMetricSuffixes {
+		mainUnitSuffix, perUnitSuffix := buildUnitSuffixes(metric.Unit())
+		if mainUnitSuffix != "" {
+			metricName = metricName + "_" + mainUnitSuffix
+		}
+		if perUnitSuffix != "" {
+			metricName = metricName + "_" + perUnitSuffix
+		}
+
+		// Append _total for Counters
+		if metric.Type() == pmetric.MetricTypeSum && metric.Sum().IsMonotonic() {
+			metricName = metricName + "_total"
+		}
+
+		// Append _ratio for metrics with unit "1"
+		// Some OTel receivers improperly use unit "1" for counters of objects
+		// See https://github.com/open-telemetry/opentelemetry-collector-contrib/issues?q=is%3Aissue+some+metric+units+don%27t+follow+otel+semantic+conventions
+		// Until these issues have been fixed, we're appending `_ratio` for gauges ONLY
+		// Theoretically, counters could be ratios as well, but it's absurd (for mathematical reasons)
+		if metric.Unit() == "1" && metric.Type() == pmetric.MetricTypeGauge {
+			metricName = metricName + "_ratio"
+		}
+	}
+	return metricName
+}
+
+func buildUnitSuffixes(unit string) (mainUnitSuffix, perUnitSuffix string) {
+	// Split unit at the '/' if any
+	unitTokens := strings.SplitN(unit, "/", 2)
+
+	if len(unitTokens) > 0 {
+		mainUnitOTel := strings.TrimSpace(unitTokens[0])
+		if mainUnitOTel != "" && !strings.ContainsAny(mainUnitOTel, "{}") {
+			mainUnitSuffix = cleanUpUnit(unitMapGetOrDefault(mainUnitOTel))
+		}
+
+		if len(unitTokens) > 1 && unitTokens[1] != "" {
+			perUnitOTel := strings.TrimSpace(unitTokens[1])
+			if perUnitOTel != "" && !strings.ContainsAny(perUnitOTel, "{}") {
+				perUnitSuffix = perUnitMapGetOrDefault(perUnitOTel)
```

Comment:
I forgot to mention this bug in the previous review, cleanUpUnit still needs to be called here:
```suggestion
				perUnitSuffix = cleanUpUnit(perUnitMapGetOrDefault(perUnitOTel))
```